opentelemetry/proto/logs/v1/logs.proto

```diff
@@ -0,0 +1,130 @@
+    // If this field is not set then no library info is known.
+    opentelemetry.proto.common.v1.InstrumentationLibrary instrumentation_library = 1;
+
+    // A list of log records.
+    repeated LogRecord logs = 2;
+}
+
+// Possible values for LogRecord.SeverityNumber.
+enum SeverityNumber {
+    UNDEFINED_SEVERITY_NUMBER = 0;
+    TRACE  = 1;
+    TRACE2 = 2;
+    TRACE3 = 3;
+    TRACE4 = 4;
+    DEBUG  = 5;
+    DEBUG2 = 6;
+    DEBUG3 = 7;
+    DEBUG4 = 8;
+    INFO   = 9;
+    INFO2  = 10;
+    INFO3  = 11;
+    INFO4  = 12;
+    WARN   = 13;
+    WARN2  = 14;
+    WARN3  = 15;
+    WARN4  = 16;
+    ERROR  = 17;
+    ERROR2 = 18;
+    ERROR3 = 19;
+    ERROR4 = 20;
+    FATAL  = 21;
+    FATAL2 = 22;
+    FATAL3 = 23;
+    FATAL4 = 24;
+}
+
+// Masks for LogRecord.flags field.
+enum LogRecordFlags {
+    UNDEFINED_LOG_RECORD_FLAG = 0;
+    TRACE_FLAGS_MASK = 0x000000FF;
+}
+
+// A log record according to OpenTelemetry Log Data Model:
+// https://github.com/open-telemetry/oteps/blob/master/text/logs/0097-log-data-model.md
+message LogRecord {
+    // time_unix_nano is the time when the event occurred.
+    // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+    // Value of 0 indicates unknown or missing timestamp.
+    fixed64 time_unix_nano = 1;
+
+    // A unique identifier for a trace. All logs from the same trace share
+    // the same `trace_id`. The ID is a 16-byte array. An ID with all zeroes
+    // is considered invalid. Can be set for logs that are part of request processing
+    // and have an assigned trace id. Optional.
+    bytes trace_id = 2;
+
+    // A unique identifier for a span within a trace, assigned when the span
+    // is created. The ID is an 8-byte array. An ID with all zeroes is considered
+    // invalid. Can be set for logs that are part of a particular processing span.
+    // If span_id is present trace_id SHOULD be also present. Optional.
+    bytes span_id = 3;
+
+    // Optional flags, a bit field. 8 least significant bits are the trace flags as
+    // defined in W3C Trace Context specification. 24 most significant bits are reserved
+    // and must be set to 0. Readers that must not assume that 24 most significant bits
+    // will be zero and must correctly mask the bits when reading 8-bit trace flag (use
+    // flags & TRACE_FLAGS_MASK). Optional.
+    fixed32 flags = 4;
+
+    // Numerical value of the severity, normalized to values described Log Data Model.
+    // Optional.
+    SeverityNumber severity_number = 5;
+
+    // The severity text (also known as log level). The original string representation as
+    // it is known at the source. Optional.
+    string severity_text = 6;
+
+    // Short event identifier that does not contain varying parts. Name describes
+    // what happened (e.g. "ProcessStarted"). Recommended to be no longer than 50
+    // characters. Not guaranteed to be unique in any way. Optional.
```

Comment:
Q: is the plan to have the proto serve as the de-facto spec for logging data, or will there be a separate document in the specification that explains the difference between `name` and `body`? For example, if I were to write an adapter for some standard logging API like Go's or SLF4J, how would I map `logger.info("hello")`? Is "hello" a name or a body?

Reply:
Yes, we have a separate doc that has more details: https://github.com/open-telemetry/oteps/blob/master/text/logs/0097-log-data-model.md
The doc includes sample mappings to help clarify how the fields are intended to be used.

Reply:
Will it be translated into spec? I view OTEPs more as WIP, regular user shouldn't be expected to read through them.

Reply:
Sure, I can do that.

Reply:
Done: https://github.com/open-telemetry/opentelemetry-specification/pull/694